### ISSUE TITLE
rustlings 2.0

### ISSUE THREAD

**Issue by oe** [rustlings/rustlings#80]
Hello! There's been talks about taking this project in a new direction, and I'm excited to say that we have initial plans to move things forward.

A big inspiration for how to make a curriculum that is self-guided, but held together by a single thread is [RubyKoans](http://rubykoans.com/), which provides you with a .zip file containing a bunch of test-driven exercises that introduce you to basic Ruby syntax and concepts. It'd be nice if Rustlings could function the same way, ideally this would be the optimal workflow:

1. Install Rust
2. Download the exercises
3. Everything else should be explained within the thing you download

This would make the project accessible to a lot more people, by making everything self-contained and available offline. Another goal should be to define a formal curriculum. Fortunately, we can probably take inspiration from Rust by Example and other similar projects that already exist, so I don't think this'll be that much of a problem.

Finally, I want to make this a cool project to contribute to! Let's work together to make this possible :star: 

There's some things we can do right now:

- Go through the existing list of issues and sort out the ones that are irrelevant right now or will be irrelevant in the future
- Introduce some consistent branding
- Find more people! There's an IRC channel now: #rustlings on Mozilla IRC
- Lay out plans for a website and update documentation to reflect the new approach
- Figure out a new curriculum

(roughly sorted in order of important/priority)

If anyone has any suggestions, or wants to contribute or help out, please throw it at me in this issue! 

Link to join IRC: https://client00.chat.mibbit.com/?server=irc.mozilla.org&channel=#rustlings

**Comment by chrisshyi** [comment]
By #rustlings do you mean #rust-beginners?

**Comment by oe** [comment]
@chrisshyi nope, the channel's called #rustlings!

**Comment by chrisshyi** [comment]
@oe Cool, would you mind posting a link for the channel? Thanks!

**Comment by oe** [comment]
@chrisshyi added in the original post!

**Comment by chrisshyi** [comment]
@oe Thanks Olivia!

**Comment by nackjicholson** [comment]
Hello,

I'm interested in getting involved with this project. I really like the direction laid out above for rustlings 2.0. I am a rustling myself, so perhaps the most effective way I can help is as a test subject for the new direction. As I advance I'm sure I could help remake some of the less advanced modules.

I will try to catch you in the IRC channel as well.

I tried to download this repo and use it like I would ruby/python koans, and I found out quickly that it wasn't made for that use case. I had no idea how to get started or compile a chunk of the code in isolation. Now that I've gone through some of the rustlings playground exercises it is clear to me that they are very compiler driven, and that is quite different from the test-driven koans in dynamic languages.

Cheers,
Will
